src/Microsoft.Extensions.Logging.Abstractions/Internal/CompositeScope.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging.Abstractions.Internal
+{
+    public class CompositeScope : IDisposable
+    {
+        private bool _isDisposed;
+
+        private IDisposable _disposable0;
+        private IDisposable _disposable1;
+        private readonly IDisposable[] _disposable;
+
+        public CompositeScope(int count)
+        {
+            if (count > 2)
+            {
+                _disposable = new IDisposable[count - 2];
+            }
+        }
+
+        public void SetDisposable(int index, IDisposable disposable)
+        {
+            if (index == 0)
+            {
+                _disposable0 = disposable;
+            }
+            else if (index == 1)
+            {
+                _disposable1 = disposable;
+            }
+            else
+            {
+                _disposable[index - 2] = disposable;
```

Comment:
check index or it will nullref is constructor had 1 or 2 and `SetDisposable` is called for index >2